packages/annotations/src/store/index.js

```diff
@@ -13,12 +13,19 @@ import * as actions from './actions';
 /**
  * Module Constants
  */
-const MODULE_KEY = 'core/annotations';
+const STORE_NAME = 'core/annotations';
 
-const store = registerStore( MODULE_KEY, {
+/**
+ * Store definition for the annotations namespace.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/master/packages/data/README.md#createStoreDefinition
+ *
+ * @type {Object}
+ */
+export const storeDefinition = createStoreDefinition( STORE_NAME );
+
+export default registerStore( STORE_NAME, {
```

Comment:
Just checking my understanding: this code should eventually morph into:
```js
const storeDefinition = createStoreDefinition( NAME, config );
const store = registerStore( storeDefinition );
```
Is that right?

Then, what is the difference between `storeDefinition` and `store`? Isn't it the same object? The `storeDefinition.__internal` already contains the full store, doesn't it? It's not just name.

Reply:
Yes, this is what @youknowriad proposed. The plan is to have a wrapping object so we don't expose Redux object directly but rather hide it and leave some space for other types of stores like the one proposed in #26866. In that, Redux store might be very special case and we don't use `__internal` for all other types of stores.

Reply:
Also, if we start pass the `storeDefinition` imported from a package to `select()` or `dispatch()`, do we still need the registry for anything? The stores are now a decentralized network.

Currently, the registry still does two things where it needs to know about all stores:
- maintaining the aggregate subscription from all stores, so that you don't need to subscribe to each store individually
- providing the `core/data` store that acts as a proxy to access individual stores "meta" resolution state

Is there anything else? And are the two features above considered legacy, or do they still have a bright future?

Reply:
@jsnajdr I think both features you mentioned could easily go away. What I'm unsure about is the notion of parent/child registries.